### ISSUE TITLE
Can't split variable if clause has a where-block

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#1891]
``` agda
data D : Set where
  c : D

test : (x : D) → D
test x = {!x!}  -- C-c C-c
  where
   y = c

-- Splitting on x reports:
-- Not a (splittable) variable: x
-- when checking that the expression ? has type D
```

**Comment by UlfNorell** [comment]
Isn't that reasonable (although I guess the error could be better)? What should it do when there are multiple patterns?

**Comment by andreasabel** [comment]
Well, if there is not exactly one clause after splitting, it could still report a (better) error.  There is no need in reporting an error just because something **could** go wrong.

**Comment by andreasabel** [comment]
The reason for the funny error is that when we have a `where` clause, we are actually in a local module on the rhs, so the free variable computation produces garbage:

``` haskell
    -- Get number of module parameters.  These cannot be split on.
    fv <- getCurrentModuleFreeVars
    let numSplittableVars = n - fv
```

**Comment by andreasabel** [comment]
After all this was easy to fix, by taking

``` haskell
fv <- getDefFreeVars f
```

instead. Made possible by the fix of #289.
